tasks/phylogenetic_inference/task_lyveset.wdl

```diff
@@ -0,0 +1,146 @@
+    read2_array_len=$(echo "${#read2[@]}")
+
+    if [ "$read1_array_len" -ne "$read2_index_array_len" ]; then
+      echo "read1 array (length: $read1_array_len) and read2 index array (length: $read2_array_len) are of unequal length." >&2
+      exit 1
+    fi
+
+    # create lyvset project
+    set_manage.pl --create ~{dataset_name}
+
+    #shuffle paired end reads
+    for index in ${!read1_array[@]}; do
+      cp ${read1_array[$index]} . && cp ${read2_array[$index]} . # move reads to cwd
+    done
+
+    shuffleSplitReads.pl --numcpus ~{cpu} -o ./interleaved *.fastq.gz 
+
+    # then moved into your project dir
+    mv ./interleaved/*.fastq.gz ~{dataset_name}/reads/
+    
+    # cleanup
+    rmdir interleaved
+    mkdir ~{dataset_name}/ref/
+    cp ~{reference_genome} ~{dataset_name}/ref/reference.fasta
+    launch_set.pl --numcpus ~{cpu} \
+    --allowedFlanking ~{allowedFlanking} \
+    --min_alt_frac ~{min_alt_frac} \
+    --min_coverage ~{min_coverage} \
+    ~{'--presets ' + presets} \
+    ~{true='--mask_phages' false='' mask_phages} \
+    ~{true='--mask_cliffs' false='' mask_cliffs} \
+    ~{true='--nomatrix' false='' nomatrix} \
+    ~{true='--nomsa' false='' nomsa} \
+    ~{true='--notrees' false='' notrees} \
+    ~{true='--fast' false='' fast} \
+    ~{true='--downsample' false='' downsample} \
+    ~{true='--sample_sites' false='' sample_sites} \
+    ~{'--read_cleaner ' + read_cleaner} \
+    ~{'--mapper ' + mapper} \
+    ~{'--snpcaller ' + snpcaller} \
+     -ref ~{dataset_name}/ref/reference.fasta ~{dataset_name}
+    
+    # rename tree file to nwk file ending
+    if [ -f ~{dataset_name}/msa/out.RAxML_bipartitions ]; then
+      mv ~{dataset_name}/msa/out.RAxML_bipartitions ~{dataset_name}/msa/out.RAxML_bipartitions.nwk
+    fi
+
+  >>>
+  output {
+    String lyveset_docker_image = docker_image
+    File? lyveset_pairwise_matrix = "~{dataset_name}/msa/out.pairwiseMatrix.tsv"
+    File? lyveset_raxml_tree = "~{dataset_name}/msa/out.RAxML_bipartitions.nwk"
+    File? lyveset_pooled_snps_vcf = "~{dataset_name}/msa/out.pooled.snps.vcf.gz"
+    File? lyveset_filtered_matrix = "~{dataset_name}/msa/out.filteredMatrix.tsv"
+    File? lyveset_alignment_fasta = "~{dataset_name}/msa/out.aln.fas"
+    File? lyveset_reference_fasta = "~{dataset_name}/reference/reference.fasta"
+    File? lyveset_masked_regions = "~{dataset_name}/reference/maskedRegions.bed"
+    Array[File]? lyveset_msa_outputs = glob("~{dataset_name}/msa/out*")
+    Array[File]? lyveset_log_outputs = glob("~{dataset_name}/log/*")
+    Array[File]? lyveset_reference_outputs = glob("~{dataset_name}/reference/*")
+    Array[File]? lyveset_bam_outputs = glob("~{dataset_name}/bam/*.bam*")
+    Array[File]? lyveset_vcf_outputs = glob("~{dataset_name}/vcf/*.vcf*")
+    File lyveset_log = stdout()
+  }
+  runtime {
+    docker: docker_image
+    memory: "~{memory} GB"
+    cpu: cpu
+    disks: "local-disk ~{disk_size} SSD"
+    preemptible: 0
+    maxRetries: 0
```

Comment:
I have not encountered any OOM errors so far, but we could increase maxRetries to 1 or 2 to enable the memory retry feature on Terra